common/src/main/java/io/druid/concurrent/Execs.java

```diff
@@ -35,24 +37,52 @@
  */
 public class Execs
 {
-  public static ExecutorService singleThreaded(String nameFormat)
+  private static int nullToNormal(@Nullable Integer priority)
```

Comment:
I think it would greatly simplify the code if we removed this method and just defaulted to null, only setting the priority if it is actually set to something, rather than assuming the default is normal.